source/_plugins/snippet.rb

```diff
@@ -41,7 +41,7 @@ def render(context)
         if @lang
           snippet = "\r\n" + "```" + @lang + "\r\n" + snippet + "\r\n" + "```" + "\r\n"
         else
-          snippet = escape(snippet)
+          snippet = "<p>" + escape(snippet) + "</p>"
```

Comment:
I think we must disable escaping, see example at the bottom
http://struts.apache.org/tag-developers/elseif-tag.html

Reply:
I was aware about this issue at first. It is not because of above `escape` call. My escape method escapes Jekyll's staffs only (e.g. `{` or `{ % raw`).

This issue is because they are escaped in our java docs. I could un-escape them in `snippet.rb` but I prefer to fix it in our .java files as improves their readability. what do you think?

Reply:
Right ... hm ... maybe as you gained another Ruby badge you would be able to prepare a plugin which will inline snippets directly into pages, like a pre-processor - not sure if this is possible at all with Jekyll.

Reply:
I'll see but before, using `snippet.rb`, now I know that .java files which have been escaped xml tags inside their javadocs. Do you agree un-escaping them to fix above issue and also improve their javadocs readability as a PR? or no, you prefer to doing it inside `snippet.rb` on site generation?

Reply:
I wouldn't change anything in a Java source code, if you can fix that in site generation it would be good. We must keep the Java source as is to allow properly generate [JavaDocs](http://struts.apache.org/maven/struts2-core/apidocs/index.html)

Reply:
Oh... thank you, I was not aware about JavaDocs site ... I fixed that in #22.

I think I also can write a tool from `snippet.rb` to replace all of snippets with their hard-coded content in md files when needed :)

Reply:
That would be osm!

Reply:
Cool :) Are all of converted Confluence's docs available in this repo or I wait for end of transition? Of course, I starts a local test and then will provide a PR when I was sure that `snippet.rb` can convert all Confluence's snippets as best as possible :)

Reply:
I think all, or at least the most important things. The rest can be migrated once we will figure out how to handle a release process and security releases (we can hide access to some pages using Confluence).